pumpkin-inventory/src/container_click.rs

```diff
@@ -0,0 +1,155 @@
+use pumpkin_world::item::ItemStack;
+
+pub struct Click {
+    pub slot: Slot,
+    pub click_type: ClickType,
+}
+
+impl Click {
+    pub fn new(mode: u8, button: i8, slot: i16) -> Self {
+        match mode {
+            0 => Self::new_normal_click(button, slot),
+            // Both buttons do the same here, so we omit it
+            1 => Self::new_shift_click(slot),
+            2 => Self::new_key_click(button, slot),
+            3 => Self {
+                click_type: ClickType::CreativePickItem,
+                slot: Slot::Normal(slot.try_into().unwrap()),
+            },
+            4 => Self::new_drop_item(button),
+            5 => Self::new_drag_item(button, slot),
+            6 => Self {
+                click_type: ClickType::DoubleClick,
+                slot: Slot::Normal(slot.try_into().unwrap()),
+            },
```

Comment:
I wonder if we may should replace 0..6 with an enum. Would be may more cool for plugin api in the future

Reply:
This whole file is about converting the packets to enums 👍